junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/CsvArgumentsProvider.java

```diff
@@ -54,22 +69,35 @@ public void accept(CsvSource annotation) {
 		Preconditions.condition(this.annotation.value().length > 0 ^ textBlockDeclared,
 			() -> "@CsvSource must be declared with either `value` or `textBlock` but not both");
 
-		String[] lines;
 		if (textBlockDeclared) {
-			lines = NEW_LINE_REGEX.split(this.annotation.textBlock(), 0);
-		}
-		else {
-			lines = this.annotation.value();
+			return parseTextBlock(this.annotation.textBlock()).stream().map(Arguments::of);
 		}
 
-		AtomicInteger index = new AtomicInteger(1);
+		AtomicInteger index = new AtomicInteger(0);
 		// @formatter:off
-		return Arrays.stream(lines)
-				.map(line -> parseLine(line, index.getAndIncrement()))
+		return Arrays.stream(this.annotation.value())
+				.map(line -> parseLine(line, index.incrementAndGet()))
```

Comment:
Just a question, what's the reason for switching to `incrementAndGet()`?

Reply:
The only reason was to be consistent with the use of the `AtomicInteger` in `parseTextBlock()`.

Otherwise, there's no overwhelming reason.